pytorch_lightning/loops/optimizer/optimizer_loop.py

```diff
@@ -159,6 +161,10 @@ def _run_optimization(
             # if no result, user decided to skip optimization
             # otherwise update running loss + reset accumulated loss
             # TODO: find proper way to handle updating running loss
+            assert self.trainer.fit_loop is not None
+            assert self.trainer.fit_loop.epoch_loop is not None
+            assert self.trainer.fit_loop.epoch_loop.batch_loop is not None
+            assert result.loss is not None
             self.trainer.fit_loop.epoch_loop.batch_loop._update_running_loss(result.loss)
```

Comment:
reviewer: this looks ridiculous I know. if you have suggestions how to get around this please let me know. Related to that is the TODO above, which is discussed here: #9220